products/pubsub/api.yaml

```diff
@@ -144,3 +144,23 @@ objects:
 
           If the subscriber never acknowledges the message, the Pub/Sub system
           will eventually redeliver the message.
+      - !ruby/object:Api::Type::String
+        name: 'messageRetentionDuration'
+        default_value: '604800s'
+        description: |
+          How long to retain unacknowledged messages in the subscription's
+          backlog, from the moment a message is published. If
+          retainAckedMessages is true, then this also configures the retention
+          of acknowledged messages, and thus configures how far back in time a
+          subscriptions.seek can be done. Defaults to 7 days. Cannot be more
+          than 7 days (`"604800s"`) or less than 10 minutes (`"600s"`).
+
+          A duration in seconds with up to nine fractional digits, terminated
+          by 's'. Example: `"600.5s"`.
+      - !ruby/object:Api::Type::Boolean
+        name: 'retainAckedMessages'
+        description: |
+          Indicates whether to retain acknowledged messages. If `true`, then
```

Comment:
I assume this defaults to false, but I think it would be valuable to be explicit for users.

Reply:
I'm 👎, false defaults should remain unspecified and the description implies that it needs to be set to true imo.

Reply:
I look at it from the perspective of describing that there is behavior of retaining or purging ack'ed messages, in which case the default behavior is only clear for people used to streaming queues. And IMO there's no harm in being explicit over implicit

Reply:
For those who are familiar with Go/Terraform boolean behavior or with streaming providers you are right it's implied.